tools/preflight/preflight_test.go

```diff
@@ -234,6 +241,414 @@ func preflightFuncsTestcases() {
 		})
 	})
 
+	Describe("Preflight volume snapshot resources test scenarios", func() {
+
+		Context("When creating source pod from pvc", func() {
+			var (
+				nameSuffix string
+				err        error
+				pvc        = &corev1.PersistentVolumeClaim{}
+				pod        = &corev1.Pod{}
+				pvcKey     types.NamespacedName
+				podKey     types.NamespacedName
+				resultChan = make(chan error, 1)
+				once       sync.Once
+			)
+
+			BeforeEach(func() {
+				once.Do(func() {
+					nameSuffix, err = CreateResourceNameSuffix()
+					pvcKey = types.NamespacedName{
+						Name:      testPVC + "-" + internal.GenerateRandomString(6, false),
+						Namespace: installNs,
+					}
+					podKey = types.NamespacedName{
+						Name:      SourcePodNamePrefix + nameSuffix,
+						Namespace: installNs,
+					}
+
+					Expect(err).To(BeNil())
+					err = createTestPVC(pvcKey)
+					Expect(err).To(BeNil())
+					Eventually(func() error {
+						return testClient.RuntimeClient.Get(ctx, pvcKey, pvc)
+					}, timeout, interval).Should(BeNil())
+
+					// Bind the PVC
+					pvc.Status.Phase = corev1.ClaimBound
+					Expect(testClient.RuntimeClient.Status().Update(ctx, pvc)).To(BeNil())
+					Eventually(func() bool {
+						Expect(testClient.RuntimeClient.Get(ctx, pvcKey, pvc)).To(BeNil())
+						return pvc.Status.Phase == corev1.ClaimBound
+					})
+
+					// create pod from pvc
+					go func(pvcName string) {
+						var testErr error
+						pod, testErr = runOps.createSourcePodFromPVC(ctx, nameSuffix, pvcName, testClient.ClientSet)
```

Comment:
move this inside `It` of whichever test requiring this. Function under test should be written & called from `It`.